packages/flutter/lib/src/material/list_tile.dart

```diff
@@ -424,10 +424,14 @@ class ListTile extends StatelessWidget {
     ));
 
     if (trailing != null) {
-      children.add(new Container(
-        margin: const EdgeInsets.only(left: 16.0),
-        alignment: FractionalOffset.centerRight,
-        child: trailing,
+      children.add(IconTheme.merge(
+        data: new IconThemeData(color: _iconColor(theme, tileTheme)),
```

Comment:
Might be worth factoring out `new IconThemeData(color: _iconColor(theme, tileTheme)),` into a value that you use for leading and trailing (which would only be initialized if either is non-null, obviously).